atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/watch/ResilientLockWatchEventCache.java

```diff
@@ -74,11 +74,15 @@ synchronized RuntimeException handleException(InvocationTargetException rethrow)
         } catch (TransactionLockWatchFailedException e) {
             throw e;
         } catch (Throwable t) {
-            log.warn("Unexpected failure occurred when trying to use the default cache. Switching to the fallback "
-                    + "implementation", t);
-            fallbackCacheSelectedCounter.inc();
-            delegate = fallbackCache;
-            throw new TransactionLockWatchFailedException("Unexpected failure in the lock watch cache", t);
+            if (delegate.equals(fallbackCache)) {
```

Comment:
are you sure you want `equals` here and not `==`?

Reply:
yeah, good catch. `equals` here would mean we never bother actually falling back, but we probably want to do that even if this was configured with the same (but not same in memory) for each for some reason